EKrut-Server/src/ekrut/server/managers/ServerShipmentManager.java

```diff
@@ -1,53 +1,148 @@
 package ekrut.server.managers;
-import java.time.LocalDateTime;
 
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import ekrut.entity.Order;
+import ekrut.entity.OrderStatus;
+import ekrut.entity.OrderType;
+import ekrut.net.ResultType;
 import ekrut.net.ShipmentRequest;
 import ekrut.net.ShipmentResponse;
+import ekrut.server.db.DBController;
+import ekrut.server.db.OrderDAO;
 import ekrut.server.intefaces.ShipmentManagerUtils;
 
+/**
+ * The {@code ServerShipmentManager} class is responsible for managing the
+ * server shipping requests. This includes fetching shipment requests for
+ * approval, confirming the shipment, confirming the delivery, and marking the
+ * order as done.
+ * 
+ * @author Nir Betesh
+ */
 public class ServerShipmentManager {
-	
+
+	OrderDAO orderDAO;
+
+	/**
+	 * 
+	 * Constructs a new {@code ServerShipmentManager} object.
+	 * 
+	 * @param con the database controller to use for accessing the database
+	 */
+	public ServerShipmentManager(DBController con) {
+		orderDAO = new OrderDAO(con);
+	}
+
+	// Q.Nir , TBD - Need to change after Yovel implementation?
+	// PLEASE, HOW CAN I DO IT?
+	// C.Nir - Message need to be sent after confirmation.
+
+	/**
+	 * Fetches a list of shipment requests for approval.
+	 * 
+	 * @param shipmentRequest the {@code ShipmentRequest} object containing the request details
+	 * @return a {@code ShipmentResponse} object with the result of the operation
+	 * @throws IllegalArgumentException if {@code shipmentRequest} is {@code null}
+	 */
+	public ShipmentResponse fetchShipmentRequests(ShipmentRequest shipmentRequest) {
+		if (shipmentRequest == null)
+			throw new IllegalArgumentException("null shipmentRequest was provided.");
+
+		ArrayList<Order> ShippingForAppoval = orderDAO.fetchOrderShipmentList(shipmentRequest.getArea());
+		if (ShippingForAppoval == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		return new ShipmentResponse(ResultType.OK);
```

Comment:
You're not returning the list of orders in the response.

Reply:
ok, so what should I do?
should I returning the list with out the response?
Or create another method sendToClient?

Reply:
I don't understand. Just put the list inside the response.